### PR TITLE
Np 4298 delete affiliation - initial

### PR DESCRIPTION
**Pull request by svenngjeto**
Handler skeleton setup, some common parameter verification and error handling

### CODE REVIEW

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     * @param input       The input object to the method. Usually a deserialized json.
+     * @param requestInfo Request headers and path.
+     * @param context     the ApiGateway context.
+     * @return the Response body that is going to be serialized in json
+     * @throws ApiGatewayException all exceptions are caught by writeFailure and mapped to error codes through the
+     *                             method {@link RestRequestHandler#getFailureStatusCode}
+     */
+    @Override
+    protected Void processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        AccessUtils.validateIdentificationNumberAccess(requestInfo);
```
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java:49]: that or `requesterIsUserAdministrator` ?

Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java:49]:
````
private void validateHasAccessRights(RequestInfo requestInfo) throws ForbiddenException, UnauthorizedException {
    if (!AccessUtils.requesterIsUserAdministrator(requestInfo)) {
        throw new ForbiddenException();
    }
}
````

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Define the success status code.
+     *
+     * @param input  The request input.
+     * @param output The response output
+     * @return the success status code.
+     */
+    @Override
+    protected Integer getSuccessStatusCode(Void input, Void output) {
+        return HttpURLConnection.HTTP_NO_CONTENT;
```
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java:68]: 200 or 204? Cristin uses 200

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java [hunk 3]
```diff
@@ -52,6 +52,8 @@
     public static final String FULL = "full";
     public static final String PERSON_ID = "id";
     public static final String PERSONS_PATH = "persons";
+    public static final String EMPLOYMENT_ID = "employmentid";
```
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java:59]: uppercase in id

**Review by torarnet — COMMENTED** [review]
See comments, otherwise okay

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java (hunk 1 above)
Comment (svenngjeto) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java:49]: Oh yes, requester must be user administrator

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java (hunk 2 above)
Comment (svenngjeto) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/handler/DeletePersonEmploymentHandler.java:68]: Service doesn't return anything -> 204 (for now...)

File: cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java (hunk 3 above)
Comment (svenngjeto) [reply cristin-commons/src/main/java/no/unit/nva/cristin/model/Constants.java:59]: Sticky shift...

**Pull request merged**